python/pyspark/sql/dataframe.py

```diff
@@ -706,6 +706,25 @@ def explain(
         assert self._sc._jvm is not None
         print(self._sc._jvm.PythonSQLUtils.explainString(self._jdf.queryExecution(), explain_mode))
 
+    def debugCodegen(self) -> None:
+        """For debugging purposes, prints to stdout all the generated code found
+        in this :class:`DataFrame` plan (i.e., the output of each WholeStageCodegen subtree).
+
+        .. versionadded:: 3.4.0
```

Comment:
Should be 3.5.0

Reply:
Thanks @HyukjinKwon, updated to 3.5.0 as suggested